pkg/dns/azure/client/client.go

```diff
@@ -141,45 +144,49 @@ func (c *recordSetClient) Delete(ctx context.Context, zone Zone, arec ARecord) e
 }
 
 type privateRecordSetClient struct {
-	client privatedns.RecordSetsClient
+	client *armprivatedns.RecordSetsClient
 }
 
-func newPrivateRecordSetClient(config Config, userAgentExtension string) (*privateRecordSetClient, error) {
-	authorizer, err := getAuthorizerForResource(config)
+func newPrivateRecordSetClient(config Config) (*privateRecordSetClient, error) {
+	_, cred, err := getAuthorizerForResource(config)
 	if err != nil {
 		return nil, err
 	}
 
-	prc := privatedns.NewRecordSetsClientWithBaseURI(config.Environment.ResourceManagerEndpoint, config.SubscriptionID)
-	prc.AddToUserAgent(userAgentExtension)
```

Comment:
The new client doesn't allow setting user-agent?  That's unfortunate, as it could make diagnosing issues more difficult, but we can live with the change.